tests/test_v4_reliability.py

```
@@ -0,0 +1,442 @@
+        ws2 = make_isolated_workspace(tmp_path, "as_sess2", "aj_job1")
+        assert ws1 != ws2
+
+    def test_raw_id_not_in_path(self, tmp_path: Path):
+        from agent.job_manager import make_isolated_workspace
+        ws = make_isolated_workspace(tmp_path, "as_rawtest", "aj_rawjob")
+        assert "as_rawtest" not in str(ws)
+        assert "aj_rawjob" not in str(ws)
+
+    def test_traversal_in_session_id_rejected(self, tmp_path: Path):
+        from agent.job_manager import make_isolated_workspace
+        with pytest.raises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)